routes/user.js

```diff
@@ -182,4 +218,34 @@ router.route("/changePassword/:id")
         });
     });
 
+router.route("/approve/:approvalS")
+    .all(function (req, res) {
+        var makeLink = function (length) {
+            var result = '';
+            var characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+            var charactersLength = characters.length;
+            for (var i = 0; i < length; i++) {
+                result += characters.charAt(Math.floor(Math.random() * charactersLength));
+            }
+            return result;
+        };
+
+        const approvalString = req.params.approvalS;
+        if (approvalString.length !== 24) return res.send(401);
+        User.findOne({where: {approvingHash: approvalString}}).then(function (user) {
```

Comment:
What if multiple people get same approval string? I know it is a very low chance, but than you would be activating multiple accounts simultaneously right?